src/SDL_ttf.c

```diff
@@ -3358,13 +3377,35 @@ static bool CollectGlyphsFromFont(TTF_Font *font, const char *text, size_t lengt
     hb_buffer_add_utf8(hb_buffer, text, (int)length, 0, -1);
     hb_buffer_guess_segment_properties(hb_buffer);
 
-    hb_feature_t userfeatures[1];
-    userfeatures[0].tag = HB_TAG('k','e','r','n');
-    userfeatures[0].value = font->enable_kerning;
-    userfeatures[0].start = HB_FEATURE_GLOBAL_START;
-    userfeatures[0].end = HB_FEATURE_GLOBAL_END;
+    if (font->hb_variations) {
+#if 0
+        //! hb_font_set_variations does nothing
+        //! hb_font_set_variation doesn't work either
+        // is it supposed to work? Is it a bug in HarfBuzz or does it not work
+        // because we're using FreeType?
+        hb_font_set_variations(font->hb_font, font->variations, font->variations_len);
+#else
+        // should we care if this returns an error? what should we do in that case?
+        FT_Set_Var_Design_Coordinates(font->face, font->ft_variations.len, font->ft_variations.coords);
+#endif
```

Comment:
Try calling [hb_ft_hb_font_changed()](https://harfbuzz.github.io/harfbuzz-hb-ft.html#hb-ft-hb-font-changed) after [hb_font_set_variations()](https://harfbuzz.github.io/harfbuzz-hb-font.html#hb-font-set-variations)?

While you no longer need to call it in HarfBuzz 11.0.0 or later, not all distributions ship this new a HarfBuzz.


```suggestion
        hb_font_set_variations(font->hb_font, font->variations, font->variations_len);
        hb_ft_hb_font_changed(font->hb_font);
```